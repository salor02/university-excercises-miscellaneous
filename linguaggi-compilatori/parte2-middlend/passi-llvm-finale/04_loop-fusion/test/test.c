#include <stdio.h>

#define N 3

void loopfusion(int a[N][N]){
    for (int i=0; i<N; i++)
        for(int j = 0; j<N; j++){
            a[j][i]=j*i;
        }
}

int main() {
    // Dichiarazione e inizializzazione della matrice a
    int a[N][N] = {{1,2,3},
                {4,5,6},
                {7,8,9}};

    loopfusion(a);

    return 0;
}

