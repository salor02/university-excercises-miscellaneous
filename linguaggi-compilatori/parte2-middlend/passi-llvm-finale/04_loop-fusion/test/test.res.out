Loop not guarded
Trip count del primo loop: 5
Trip count del secondo loop: 5	I loop iterano lo stesso numero di volte
successor 
for.inc:                                          ; preds = %for.body8, %for.body
  %inc = add nsw i32 %i.0, 1
  br label %for.cond, !llvm.loop !6
